modules/lang-painless/src/main/java/org/elasticsearch/painless/AnalyzerCaster.java

```diff
@@ -646,6 +646,13 @@ public static Cast getLegalCast(Location location, Type actual, Type expected, b
                 }
 
                 break;
+            case MATCHER:
```

Comment:
I'm no expert on regexes, so my question here is can you have a script where you do something like ```boolean x = (boolean)someMatcher;```?  If not, this isn't the correct place for this.